test/orderstatistictree.js

```diff
@@ -0,0 +1,76 @@
+const OrderStatisticTreeMock = artifacts.require("./OrderStatisticTreeMock.sol");
+
+const getRandomInt = async function (max) {
+  return Math.floor(Math.random() * Math.floor(max));
+};
+
+
+contract('OrderStatisticTree', function () {
+
+  it("insert, rank and count ", async function () {
+    var values = [3,5,5,6,1,100,50];
+    var ranks  = [1,2,2,4,0,6,5];
+    var orderStatisticTree = await OrderStatisticTreeMock.new();
+    for (var i = 0;i<values.length;i++) {
+      await orderStatisticTree.insert(values[i]);
+    }
+    var count = await orderStatisticTree.count();
+
+    var rank;
+    assert.equal(count,values.length);
+    for (i = 0 ;i< count ;i++) {
+      rank = await orderStatisticTree.rank(values[i]);
+      assert.equal(ranks[i],rank);
+    }
+
+    var valuesInBetween = [4,8,20,101,2];
+    var ranksInBetween  = [2,5,5,7,1];
```

Comment:
I recommend adding to the test a duplicate case just to cover that up as well. Maybe rank 5 or 6 to see make sure the ranking supports duplicates as well.

Reply:
The duplicate case is already covered here https://github.com/daostack/arc/blob/threshold_exclude_expiered_proposals/test/orderstatistictree.js#L11 .
Did I missed something ?

Reply:
No, sorry, I missed that one.